src/main/dialogs/quickSetStrip/qssTooltipDialog.js

```diff
@@ -157,11 +188,31 @@ gpii.app.qssTooltipDialog.showIfPossible = function (that, setting, btnCenterOff
  * the screen.
  */
 gpii.app.qssTooltipDialog.getTooltipPosition = function (that, btnCenterOffset) {
-    var arrowWidth = that.options.arrowWidth,
-        scaleFactor = that.model.scaleFactor;
+    var screen = require("electron").screen, // used to get the current screen size
+        availableDirections = that.model.availableDirections,
+        arrowDirection = availableDirections.defaultDirection, // default tooltip arrow direction
+        arrowWidth = that.options.arrowWidth,
+        scaleFactor = that.model.scaleFactor,
+        tooltipWidth = that.options.config.attrs.width * scaleFactor, // current tooltip width
+        screenWidth = screen.getPrimaryDisplay().workAreaSize.width, // current screen size
+        offsetX = btnCenterOffset.x - scaleFactor * arrowWidth / 2; // calculate the offset
+
+    // checking if the offset is too big and the tooltip will show off screen
+    if (offsetX + tooltipWidth > screenWidth) {
+        // setting the offset to fit the screen
+        offsetX = screenWidth - tooltipWidth;
+        // changing the arrow to be in the center
+        arrowDirection = availableDirections.centerDirection;
+    } else {
+        arrowDirection = availableDirections.defaultDirection;
+    }
+
+    // apply the new arrow direction
+    that.applier.change("arrowDirection", arrowDirection);
```

Comment:
This is an odd side-effect in a function whose purpose is to return the offsets structure. I'm guessing that the original intention was to implement this inside the getArrowDirectionFunction - I suggest that we go either wholesale with a model-driven approach or else continue with this getXxx approach and then instead add the arrowDirection into the returned payload from this function

Reply:
I imagine it is impossible for the tooltip direction to change during the lifetime of the displayed tooltip, right?

Reply:
Yeah, I do agree that it's strange to have applier in a get type function. Probably will be easier to just return the value and set it elsewhere, I will take a look.

And yes, I'm pretty sure the direction won't change during one instance of the tooltip.

Reply:
**Update:** I've made a slight change (your second recommendation) to where the direction is set.